### PR TITLE
CTR mode encryption

### PR DESCRIPTION
**Pull request by domsj**
- better fit for partial reads
- no padding needed for encryption

### CODE REVIEW

**Comment by toolslive** [comment]
ok.

File: ocaml/src/fragment_helper.ml [hunk 1]
```diff
@@ -104,13 +119,46 @@ let maybe_decrypt
...
+        get_iv algo key ~object_id ~chunk_id ~fragment_id ~ignore_fragment_id >>= fun iv ->
+        Cipher.with_t_lwt
+          key Cipher.AES256 Cipher.CTR []
+          (fun cipher ->
+            Cipher.set_ctr cipher iv;
+            Cipher.decrypt_detached
+                  cipher
+                  data 0 (Lwt_bytes.length data)
+          ) >>= fun () ->
+        Lwt.return (Bigstring_slice.wrap_bigstring data)
```
Comment (domsj) [inline ocaml/src/fragment_helper.ml]: Fwiw it's not only the last step (unpadding) that differs, also the chosen mode (CBC/CTR) and set_iv/set_ctr differ. Lifting the commonalities highlights the differences, so I'll do that

**Comment by domsj** [comment]
In https://github.com/openvstorage/alba/pull/439 it was discovered that the previous assumption that compression will always lead to the same output was wrong (compressors evolve).
In combination with deterministic IV/CTR generation and the CTR mode encryption implemented here this could (at least theoretically) leak information about the compressed data.
After some discussion with @toolslive about this on the chat we decided to generate a random IV/CTR upon encryption and store it in the manifest (and recovery info).

**Comment by domsj** [comment]
@toolslive I added some more changes, should be finished now. Can you review again?

**Comment by toolslive** [comment]
ok.

**Pull request merged**